### PR TITLE
Feature/33 build docs hints

### LINKED ISSUE #33: Building docs fails because of missing dependency
Following the instructions:
```bash
poetry run sphinx-apidoc -o docs/source/api src
poetry run sphinx-build -M html docs/source docs/build
```
the second command fails with
```
Running Sphinx v4.5.0
loading translations [en]... done

Extension error:
Could not import extension sphinx-favicon (exception: No module named 'sphinx-favicon')
```

### PR DESCRIPTION
**Pull request by poikilotherm**
Closes #33

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,7 +8,9 @@ For more information about the HERMES project, see the [HERMES project website](
 
 - Python sources are in the `src` folder
 - pytest tests are in the `test` folder
-- Architectural design records are in `docs/adr`
+- [Architectural Design Records (ADR)](https://adr.github.io/) are in `docs/adr`
+- API documentation is generated in `docs/source/api`
```
Comment (sdruskat) [inline README.md:12]:
```suggestion
- API documentation is generated into `docs/source/api`
```

**suggestion:** Change to *into* to make clear that the docs aren't generated from within this dir.

File: docs/source/conf.py [hunk 2]
```diff
@@ -31,6 +31,9 @@
 # extensions coming with Sphinx (named 'sphinx.ext.*') or your custom
 # ones.
 extensions = [
+    # run sphinx-apidocs on every build
+    'sphinxcontrib.apidoc',
```
Comment (sdruskat) [inline docs/source/conf.py:35]: ::question/suggestion:** I have recently switched from `apidoc` to [`autoapi`](https://sphinx-autoapi.readthedocs.io/en/latest/tutorials.html), which I found to be even less intrusive. Have you used it in the past? And if so, are there reasons not to switch?

File: pyproject.toml [hunk 3]
```diff
@@ -14,18 +16,41 @@ click = "^8.1"
...
 cff = "hermes.commands.harvest.cff:harvest_cff"
 
+[tool.taskipy.tasks]
+docs-build = "poetry run sphinx-build -M html docs/source docs/build -W"
+docs-clean = "poetry run sphinx-build -M clean docs/source docs/build"
+docs-live = "poetry run sphinx-autobuild docs/source docs/build"
+
 [build-system]
-requires = ["poetry-core>=1.0.0"]
+requires = ["poetry-core>=1.2.0"]
```
Comment (sdruskat) [inline pyproject.toml:55]: **question:** Does `taskipy` require this?

**Review by sdruskat — APPROVED** [review]
Looks good, some minor comments. Feel free to merge yourself after addressing these, @poikilotherm.

File: pyproject.toml (hunk 3 above)
Comment (poikilotherm) [reply pyproject.toml:55]: Nope, this is necessary to have proper dependency groups support. The docs group for example can now be installed with `poetry install --with docs`. This functionality became stable in v1.2 release of `poetry`. Enabling poetry use with RTD and reusing parties don't trip over this by enforcing the version.

File: docs/source/conf.py (hunk 2 above)
Comment (poikilotherm) [reply docs/source/conf.py]: Nope, nothing holding back doing so. Just my non-familiarity with this ecosystem. I migrated in [eb760d8](https://github.com/hermes-hmc/workflow/pull/48/commits/eb760d83d8dbce1b69c88d7e61ed86aa83121d2b)

Please take a look at the RTD preview if this looks good to you.

Comment (sdruskat) [reply docs/source/conf.py]: Yes, looking good, feel free to merge.

**Review by sdruskat — APPROVED** [review]
LGTM.

**Pull request merged**